### ISSUE TITLE
`/utilities/stat` `mode` return value has been broken

### ISSUE THREAD

**Issue by chrisjsewell** [eth-cscs/firecrest#171]
In https://github.com/eth-cscs/firecrest/commit/8fa4044fbe383aea88af5b74803f072a222edb79, @aledabin changed the code I added (#133) from returning `%f` to `%a` for the mode, this has broken its use 😅 

Let me explain, for a simple (empty) file:

```
$ stat -c '%a' example.txt
644
$ stat -c '%f' example.txt
81a4
```

Here `81a4` is the hexidecimal encoding of BOTH the permissions and the file type.
For example, using Python you can do:

```python
import stat
# recover the permissions as octahedral
oct(int("81a4", 16) & 0o777)[2:] == "644"
# recover the permissions as string
stat.filemode(int("81a4", 16)) == "-rw-r--r--"
# check the file type
stat.S_ISREG(int("81a4", 16)) is True
stat.S_ISDIR(int("81a4", 16)) is False
```

There is no longer any way to get the file type

as it says at: https://github.com/eth-cscs/firecrest/blob/028edf3bf492c803f4033753f80f61e6747ce248/src/utilities/utilities.py#L438
what we actually want is enough information that can be used to directly initialise https://docs.python.org/3/library/os.html#os.stat_result, you can do this with `%f` but not `%a`

---

I would also note, that `/utilities/file` is kind of useless, because for this example, it simply returns `empty`.
So no actual information on what the file type is 😬

**Comment by chrisjsewell** [comment]
I would note also, it would be great if you could construct `stat_result` from the return items of `utilities/ls`:

Currently, you can at least create the `mode` from the function:

```python
def convert_st_mode(data: dict) -> int:
    # created from `ls -l -A --time-style=+%Y-%m-%dT%H:%M:%S`, e.g.
    # -rw-------  1 username usergroup 57 2021-12-02T10:42:00 file.txt
    # type, permissions, # of links (not recorded), user, group, size, last_modified, name

    ftype = {
        "b": "0060",  # block device
        "c": "0020",  # character device
        "d": "0040",  # directory
        "l": "0120",  # Symbolic link
        "s": "0140",  # Socket.
        "p": "0010",  # FIFO
        "-": "0100",  # Regular file
    }
    p = item["permissions"]
    r = lambda x: 4 if x == "r" else 0  # noqa: E731
    w = lambda x: 2 if x == "w" else 0  # noqa: E731
    x = lambda x: 1 if x == "x" else 0  # noqa: E731
    p_int = (
        ((r(p[0]) + w(p[1]) + x(p[2])) * 100)
        + ((r(p[3]) + w(p[4]) + x(p[5])) * 10)
        + ((r(p[6]) + w(p[7]) + x(p[8])) * 100)
    )
    mode = int(ftype[item["type"]] + str(p_int), 8)

    return mode
```

**Comment by chrisjsewell** [comment]
Ah I see I was actually mentioned by @aledabin in the original PR: https://github.com/eth-cscs/firecrest/pull/133#issuecomment-1032315023

but yeh, its wrong for the reasons mentioned above; it needs to encode both permissions and file type

**Comment by chrisjsewell** [comment]
FYI, this is the actual case I had in mind for this endpoint 😄 https://github.com/eth-cscs/pyfirecrest/pull/43

**Comment by aledabin** [comment]
Hi @chrisjsewell , sorry, I misunderstood the whole issue. It will be published and deployed soon.

Regarding the `file` endpoint it returns the output of the **file** command, so if the file is empty it returns that. Could that be the reason?

**Comment by aledabin** [comment]
Hi @chrisjsewell, the fix was included in v1.8.3 and deployed at CSCS last month, I forgot to update here :disappointed: 

Regarding `file` endpoint, I understand your point, maybe it is not that useful to be exposed.

I'll close this issue.
Best,

**Comment by chrisjsewell** [comment]
No worries, cheers!
